packages/vitest/src/node/cli/cli-api.ts

```diff
@@ -132,3 +118,91 @@ export async function startVitest(
   await ctx.close()
   return ctx
 }
+
+export async function prepareVitest(
+  mode: VitestRunMode,
+  options: CliOptions = {},
+  viteOverrides?: ViteUserConfig,
+  vitestOptions?: VitestOptions,
+): Promise<Vitest> {
+  process.env.TEST = 'true'
+  process.env.VITEST = 'true'
+  process.env.NODE_ENV ??= 'test'
+
+  if (options.run) {
+    options.watch = false
+  }
+
+  // this shouldn't affect _application root_ that can be changed inside config
+  const root = resolve(options.root || process.cwd())
+
+  // running "vitest --browser.headless"
+  if (typeof options.browser === 'object' && !('enabled' in options.browser)) {
+    options.browser.enabled = true
+  }
+
+  if (typeof options.typecheck?.only === 'boolean') {
+    options.typecheck.enabled ??= true
+  }
+
+  const ctx = await createVitest(mode, options, viteOverrides, vitestOptions)
+
+  const environmentPackage = getEnvPackageName(ctx.config.environment)
+
+  if (
+    environmentPackage
+    && !(await ctx.packageInstaller.ensureInstalled(environmentPackage, root))
+  ) {
+    process.exitCode = 1
+    return ctx
+  }
+
+  return ctx
+}
+
+export function processCollected(files: File[], options: CliOptions) {
+  if (typeof options.json === 'boolean') {
+    return console.log(JSON.stringify(formatCollectedAsJSON(files), null, 2))
+  }
+
+  if (typeof options.json === 'string') {
+    const jsonPath = resolve(options.root || process.cwd(), options.json)
+    mkdirSync(dirname(jsonPath), { recursive: true })
+    writeFileSync(jsonPath, JSON.stringify(formatCollectedAsJSON(files), null, 2))
+    return
+  }
+
+  return formatCollectedAsString(files).forEach(test => console.log(test))
+}
+
+export function formatCollectedAsJSON(files: File[]) {
+  return files.map((file) => {
+    const tests = getTests(file).filter(test => test.mode === 'run' || test.mode === 'only')
+    return tests.map((test) => {
+      const result: any = {
+        name: getNames(test).slice(1).join(' > '),
+        file: file.filepath,
+      }
+      if (test.file.projectName) {
+        result.projectName = test.file.projectName
+      }
+      if (test.location) {
+        result.location = test.location
+      }
+      return result
+    })
+  }).flat()
+}
+
+export function formatCollectedAsString(files: File[]) {
```

Comment:
When running in `test/coverage-test`, it would be nice to have all tests sorted so that workspace projects would show nicely. But I guess order comes from sequencer here too?

Reply:
Yeah, I don't think it makes sense to sort them after they run. They are already sorted by that point